src/library/basetrackcache.cpp

```diff
@@ -228,10 +191,6 @@ bool BaseTrackCache::updateIndexWithQuery(const QString& queryString) {
         qDebug() << "updateIndexWithQuery issuing query:" << queryString;
     }
 
-    if (m_bIsCaching) {
-        resetRecentTrack();
```

Comment:
This was the root cause of the deadlock. This gets called in responds of another track that is evicted form cache. 
Here another track is evicted form cache leading to a recursive call.